lightning/checkpoints/checkpoints.go

```diff
@@ -741,7 +741,25 @@ func NewFileCheckpointsDB(path string) *FileCheckpointsDB {
 	// ignore all errors -- file maybe not created yet (and it is fine).
 	content, err := ioutil.ReadFile(path)
 	if err == nil {
-		cpdb.checkpoints.Unmarshal(content)
+		err2 := cpdb.checkpoints.Unmarshal(content)
+		if err2 != nil {
+			log.L().Error("checkpoint file is broken", zap.String("path", path), zap.Error(err2))
+		}
+		// FIXME: patch for empty map may need initialize manually, because currently
+		// FIXME: a map of zero size -> marshall -> unmarshall -> become nil, see checkpoint_test.go
```

Comment:
why using FIXME for these comments?

Reply:
I think it's an unexpected behaviour of `gogo/protobuf`'s marshall and unmarshall. There's logic to recover a empty map(but didn't execute) in https://github.com/pingcap/tidb-lightning/blob/master/lightning/checkpoints/file_checkpoints.pb.go#L877-L878

If `gogo/protobuf` fixed this bug, we won't bother to initiate. So I added a FIXME to remind to report bug and check if fixed from `gogo/protobuf`.